src/ray/stats/metric_exporter_client.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RAY_METRIC_EXPORTER_CLIENT_H
+#define RAY_METRIC_EXPORTER_CLIENT_H
+#include "ray/rpc/gcs_server/gcs_rpc_client.h"
+
+#include "ray/stats/metric.h"
+
+namespace ray {
+namespace stats {
+/// Interface class for abstract metrics exporter client.
+class MetricExporterClient {
+ public:
+  virtual void ReportMetrics(const MetricPoints &points) = 0;
+  virtual ~MetricExporterClient() = default;
+};
+
+/// Default stdout exporter client can log metrics info for debug.
+class StdoutExporterClient : public MetricExporterClient {
```

Comment:
Why doesn't this extend decorator?

Reply:
In decorator pattern, a basic concrete class is needed, so we use stdout as the default concrete class.

For example, one person can wear different clothes. At least one person's instance should be constructed first, then the clothes can be worn piece by piece.

Reply:
I see. That means we should always have this client configured. Am I correct? Are you going to add gRPC exporter in the following PR? (so you guys will always have at least openTSDB & gRPC (for dashboard) & stdout)

Reply:
> I see. That means we should always have this client configured. Am I correct? Are you going to add gRPC exporter in the following PR? (so you guys will always have at least openTSDB & gRPC (for dashboard) & stdout)

Yes. Dashboard exporter client or other client  can be the first level instance in pattern, but I think default stdout is better than any others, which's easy for debugging or raw metric preview.

Reply:
I see. Thanks for the response! I will unresolve this conversation just in case @edoakes has any opinion.